### PR TITLE
Reduce references to Context in browser-icons

### PR DESCRIPTION
**Pull request by NotWoods**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt [hunk 1]
```diff
@@ -79,7 +79,7 @@ class BrowserIcons(
     jobDispatcher: CoroutineDispatcher = Executors.newFixedThreadPool(THREADS).asCoroutineDispatcher()
 ) {
     private val logger = Logger("BrowserIcons")
-    private val maximumSize = context.resources.pxToDp(MAXIMUM_SIZE_DP)
+    private val maximumSize = context.resources.displayMetrics.pxToDp(MAXIMUM_SIZE_DP)
```
Comment (jhugman) [inline components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt]: This code doesn't look right. I understand that this is a fairly mechanical refactor. 

The function `pxToDp` would appear to take a `px` value and spit out a `dp` value. This line appears to give that function a `dp`.

I think `git blame` and ask whoever wrote this line originally.

File: components/support/ktx/src/main/java/mozilla/components/support/ktx/android/content/res/Resources.kt [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
 import android.content.res.Resources
-import android.util.TypedValue
+import mozilla.components.support.ktx.android.util.pxToDp
 
 /**
  * Converts a value in density independent pixels (pxToDp) to the actual pixel values for the display.
  */
-fun Resources.pxToDp(pixels: Int) = TypedValue.applyDimension(
-        TypedValue.COMPLEX_UNIT_DIP, pixels.toFloat(), displayMetrics).toInt()
+fun Resources.pxToDp(pixels: Int) = displayMetrics.pxToDp(pixels)
```
Comment (jhugman) [inline components/support/ktx/src/main/java/mozilla/components/support/ktx/android/content/res/Resources.kt]: Here is the problem. The documentation and the method name do not match. Which one is correct?

File: components/support/ktx/src/main/java/mozilla/components/support/ktx/android/util/DisplayMetrics.kt [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+package mozilla.components.support.ktx.android.util
+
+import android.util.DisplayMetrics
+import android.util.TypedValue
+
+/**
+ * Converts a value in density independent pixels (pxToDp) to the actual pixel values for the display.
+ */
+fun DisplayMetrics.pxToDp(pixels: Int) = TypedValue.applyDimension(
+    TypedValue.COMPLEX_UNIT_DIP, pixels.toFloat(), this).toInt()
```
Comment (jhugman) [inline components/support/ktx/src/main/java/mozilla/components/support/ktx/android/util/DisplayMetrics.kt]: The `TypeValue.applyDimension` docs suggest that this the method name should be `dpToPx`.

We should file an issue and fix this. 

[Units are hard][1].

[1]: http://edition.cnn.com/TECH/space/9909/30/mars.metric.02/

File: components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt (hunk 1 above)
Comment (NotWoods) [reply components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt]: #754 is where it was decided on. I'll ask about the name.

**Comment by NotWoods** [comment]
I was going to file a follow-up issue, but [turns out it already exists](https://github.com/mozilla-mobile/android-components/issues/2605).

File: components/browser/icons/src/main/java/mozilla/components/browser/icons/utils/DiskCache.kt [hunk 4]
```diff
@@ -61,11 +61,11 @@ internal class DiskCache :
         return emptyList()
     }
 
-    internal fun putResources(context: Context, request: IconRequest) {
+    internal fun putResources(cacheDir: File, request: IconRequest) {
```
Comment (pocmo) [inline components/browser/icons/src/main/java/mozilla/components/browser/icons/utils/DiskCache.kt]: As mentioned in the other PR: I do not think I agree with such a refactoring: `Context` is needed to lookup a  directory. Passing in a `File` is way more generic and requires the caller to know which File to pass in (which is meant to be an implementation detail). This doesn't protect against a caller providing wrong/other directories.

Comment (NotWoods) [reply components/browser/icons/src/main/java/mozilla/components/browser/icons/utils/DiskCache.kt]: For private/internal functions, is that an issue?

Comment (pocmo) [reply components/browser/icons/src/main/java/mozilla/components/browser/icons/utils/DiskCache.kt]: I agree that it's less an issue than a public API - but I expect many people to work on that code over time... :)

File: components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt (hunk 1 above)
Comment (dector) [reply components/browser/icons/src/main/java/mozilla/components/browser/icons/BrowserIcons.kt]: May I suggest API like:

```kotlin
private val maximumSize = MAXIMUM_SIZE_DP.pxToDpWith(context)
```

?

**Comment by NotWoods** [comment]
Closing as many of the changes landed in other PRs

**Pull request closed without merging**